File: ADA_Tema_6/Prueba_SQLiteRoom/app/src/main/java/com/example/prueba_sqliteroom/room/AppDB.java
package com.example.prueba_sqliteroom.room;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.prueba_sqliteroom.room.entidades.Artista;
import com.example.prueba_sqliteroom.room.iDaos.Artista_IDao;

@Database(entities = {Artista.class}, version = 1)
public abstract class AppDB extends RoomDatabase {
    public static AppDB INSTANCE;

    public abstract Artista_IDao daoArtista();

    public static AppDB getAppDB(Context context) {
        if (INSTANCE == null) {
            //Usar allowMainThreadQueries() solo para pruebas
            INSTANCE = Room
                    .databaseBuilder(context.getApplicationContext(), AppDB.class,
                            Constantes.Constante.NOMBRE_DB)
                    .allowMainThreadQueries()
                    .build();
        }
        return INSTANCE;
    }

    public void destroyInstance() {
        INSTANCE = null;
    }
}

File: ADA_Tema_6/Kotllin_Prueba_SQLite/app/src/main/java/com/example/kotllin_prueba_sqlite/room/AppDB.kt
package com.example.kotllin_prueba_sqlite.room

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.kotllin_prueba_sqlite.room.dao.IDAOArtista
import com.example.kotllin_prueba_sqlite.room.dao.IDAOCancion
import com.example.kotllin_prueba_sqlite.room.entities.Artista
import com.example.kotllin_prueba_sqlite.room.entities.Cancion


@Database(entities = [Artista::class,Cancion::class], version = 1)
abstract class AppDB : RoomDatabase() {
    abstract fun daoArtista(): IDAOArtista
    abstract fun daoCancion(): IDAOCancion
    fun destroyInstance() {
        INSTANCE = null
    }

    companion object {
        var INSTANCE: AppDB? = null

        fun getAppDB(context: Context): AppDB? {
            if (INSTANCE == null) {
                //Usar allowMainThreadQueries() solo para pruebas
                INSTANCE = Room
                    .databaseBuilder(
                        context.applicationContext,
                        AppDB::class.java,
                    "Spotify"
                    ).allowMainThreadQueries()
                    .build()
            }
            return INSTANCE
        }
    }
}